src/lib.rs

```diff
@@ -15,22 +15,40 @@ use syntax::feature_gate::AttributeType;
 use syntax::symbol::Symbol;
 use syntax::util::small_vector::SmallVector;
 
-pub fn insert_flame_guard(cx: &mut ExtCtxt, _span: Span, _mi: &MetaItem,
+pub fn insert_flame_guard(cx: &mut ExtCtxt, _span: Span, mi: &MetaItem,
                           a: Annotatable) -> Annotatable {
+    let opt_ident = match mi.node {
+        MetaItemKind::Word => None,
+        MetaItemKind::List(ref v) => {
```

Comment:
Could this be simplified to match against `MetaItemKind::List(ref v) if v.len() == 1 => { .. }` and `_ => None`?